src/screens/Order/OrderUIUtils.res

```diff
@@ -98,22 +132,130 @@ let filterByData = (txnArr, value) => {
   })
 }
 
-let initialFilters = json => {
+let getConditionalFilter = (key, dict, filterValues) => {
+  open LogicUtils
+
+  let filtersArr = switch key->getFilterTypeFromString {
+  | #connector_label => {
+      let arr = filterValues->getArrayFromDict("connector", [])->getStrArrayFromJsonArray
+      let new_arr = []
+      arr->Array.forEach(item => {
+        let connectorLabelArr = dict->getDictfromDict("connector")->getArrayFromDict(item, [])
+        connectorLabelArr->Array.forEach(item => {
+          let connector_label = item->getDictFromJsonObject->getString("connector_label", "")
+          new_arr->Array.push(connector_label)
+        })
+      })
+      new_arr
+    }
+  | #payment_method_type => {
+      let arr = filterValues->getArrayFromDict("payment_method", [])->getStrArrayFromJsonArray
+
+      let new_arr = []
+      arr->Array.forEach(item => {
+        let paymentMethodTypeArr =
+          dict
+          ->getDictfromDict("payment_method")
+          ->getArrayFromDict(item, [])
+          ->getStrArrayFromJsonArray
+        paymentMethodTypeArr->Array.forEach(item => {
+          new_arr->Array.push(item)
+        })
+      })
+      new_arr
+    }
+  | _ => []
+  }
+
+  filtersArr
+}
+
+let getMerchantIdforConnector: (
```

Comment:
can we reuse this function.Its getting repeated in the refund also

Reply:
renamed the function for now, we can make common utils for filters in refunds & orders later on